lib/plugin.js

```diff
@@ -71,9 +72,15 @@ ManifestPlugin.prototype.apply = function(compiler) {
       return memo;
     }, {}));
 
-    // Append optional basepath onto all references.
-    // This allows output path to be reflected in the manifest.
-    if (this.opts.basePath) {
+    if (this.opts.useBasePathAndPublicPath) {
```

Comment:
Also as an added bonus, maybe you could fix the concatenation of base and public paths to use `path.resolve(this.opts.publicPath, value)` while you're in here? The current way we're doing it is fairly fragile.